diff --git a/src/DisqusCommentManager.php b/src/DisqusCommentManager.php
index efb5443..e80e822 100644
--- a/src/DisqusCommentManager.php
+++ b/src/DisqusCommentManager.php
@@ -100,7 +100,7 @@ class DisqusCommentManager implements DisqusCommentManagerInterface {
       // The login window must be closed once the user logs in.
       'url' => Url::fromRoute('user.login', [], ['query' => ['destination' => 'disqus/closewindow'], 'absolute' => TRUE])->toString(),
       // The logout link must redirect back to the original page.
-      'logout' => Url::fromRoute('user.logout', [], ['query' => ['destination' => Url::fromRoute('<current>')], 'absolute' => TRUE])->toString(),
+      'logout' => Url::fromRoute('user.logout', [], ['query' => ['destination' => Url::fromRoute('<current>')->toString()], 'absolute' => TRUE])->toString(),
       'width' => 800,
       'height' => 600,
     ];
