diff --git a/src/Form/BaseRatingForm.php b/src/Form/BaseRatingForm.php
index e518757..d83e94d 100644
--- a/src/Form/BaseRatingForm.php
+++ b/src/Form/BaseRatingForm.php
@@ -2,7 +2,10 @@
 
 namespace Drupal\votingapi_widgets\Form;
 
+use Drupal\Component\Datetime\TimeInterface;
 use Drupal\Core\Entity\ContentEntityForm;
+use Drupal\Core\Entity\EntityRepositoryInterface;
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Component\Utility\Html;
 use Symfony\Component\DependencyInjection\ContainerInterface;
@@ -23,7 +26,8 @@ class BaseRatingForm extends ContentEntityForm {
   /**
    * Class constructor.
    */
-  public function __construct(VoteResultFunctionManager $votingapi_result) {
+  public function __construct(VoteResultFunctionManager $votingapi_result, EntityRepositoryInterface $entity_repository, EntityTypeBundleInfoInterface $entity_type_bundle_info = NULL, TimeInterface $time = NULL) {
+    parent::__construct($entity_repository, $entity_type_bundle_info, $time);
     $this->votingapiResult = $votingapi_result;
   }
 
@@ -32,7 +36,10 @@ class BaseRatingForm extends ContentEntityForm {
    */
   public static function create(ContainerInterface $container) {
     return new static(
-      $container->get('plugin.manager.votingapi.resultfunction')
+      $container->get('plugin.manager.votingapi.resultfunction'),
+      $container->get('entity.repository'),
+      $container->get('entity_type.bundle.info'),
+      $container->get('datetime.time')
     );
   }
 
