diff --git a/ds.module b/ds.module
index 4f8c590..7382e18 100644
--- a/ds.module
+++ b/ds.module
@@ -470,13 +470,12 @@ function ds_preprocess_ds_layout(&$variables) {
  */
 function ds_theme_suggestions_alter(&$suggestions, $variables, $base_theme_hook) {
   if (isset($variables['content']) && is_array($variables['content']) && isset($variables['content']['#ds_configuration']) && $base_theme_hook != 'ds_entity_view') {
-    $layout_id = $variables['content']['#ds_configuration']['layout']['id'];
     $entity_id = $variables['content']['#' . $variables['content']['#entity_type']]->id();
-    $suggestions[] = $layout_id . '__' . $variables['content']['#entity_type'];
-    $suggestions[] = $layout_id . '__' . $variables['content']['#entity_type'] . '_' . $variables['content']['#view_mode'];
-    $suggestions[] = $layout_id . '__' . $variables['content']['#entity_type'] . '_' . $variables['content']['#bundle'];
-    $suggestions[] = $layout_id . '__' . $variables['content']['#entity_type'] . '_' . $variables['content']['#bundle'] . '_' . $variables['content']['#view_mode'];
-    $suggestions[] = $layout_id . '__' . $variables['content']['#entity_type'] . '__' . $entity_id;
+    $suggestions[] = $base_theme_hook . '__' . $variables['content']['#entity_type'];
+    $suggestions[] = $base_theme_hook . '__' . $variables['content']['#entity_type'] . '_' . $variables['content']['#view_mode'];
+    $suggestions[] = $base_theme_hook . '__' . $variables['content']['#entity_type'] . '_' . $variables['content']['#bundle'];
+    $suggestions[] = $base_theme_hook . '__' . $variables['content']['#entity_type'] . '_' . $variables['content']['#bundle'] . '_' . $variables['content']['#view_mode'];
+    $suggestions[] = $base_theme_hook . '__' . $variables['content']['#entity_type'] . '__' . $entity_id;
   }
 }
 
