diff --git a/ds.module b/ds.module
index 4f8c590..ed8857f 100644
--- a/ds.module
+++ b/ds.module
@@ -693,6 +693,12 @@ function ds_theme_suggestions_field_alter(&$suggestions, $variables) {
       $suggestions[] = $suggestion . '__' . $variables['element']['#entity_type'] . '__' . $variables['element']['#bundle'];
       $suggestions[] = $suggestion . '__' . $variables['element']['#entity_type'] . '__' . $field_name . '__' . $variables['element']['#bundle'];
     }
+
+    // Custom DS fields name may contain a colon separator; replace it with "__"
+    // to ensure suggestions are compatible with file names on all systems.
+    foreach ($suggestions as $key => $suggestion) {
+      $suggestions[$key] = str_replace(":", "__", $suggestion);
+    }
   }
 }
 
