diff --git a/src/SchemaMetatagManager.php b/src/SchemaMetatagManager.php
index 77c8449..43e38a6 100644
--- a/src/SchemaMetatagManager.php
+++ b/src/SchemaMetatagManager.php
@@ -130,7 +130,7 @@ class SchemaMetatagManager implements SchemaMetatagManagerInterface {
       $value = $exploded[0];
     }
     else {
-      $value = $exploded;
+      $value = array_map('trim', $exploded);
     }
     return $value;
   }
